feat: add `streamlit` spec

jrieke opened the pull request:
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Feature

**What is the current behavior? (You can also link to an open issue here)**

No autocompletion for [Streamlit](https://streamlit.io/) 😢

**What is the new behavior (if this is a feature change)?**

Added autocompletion for Streamlit 🎈, with all commands and options mentioned in the help message: 

<img width="438" alt="CleanShot 2022-08-05 at 16 53 51@2x" src="https://user-images.githubusercontent.com/5103165/183103303-8ae0ad65-fb33-4c45-935f-0763c2883d80.png">

**Additional info:**

1. Great tool!! Just discovered 2h ago and using fig + creating a spec was super smooth.
2. I'm a PM at Streamlit, so happy to maintain this if we update something on our side. 
3. If you want to know more about Streamlit, feel free to reach out 😉

fedeci commented on src/streamlit.ts:
```diff
@@ -0,0 +1,74 @@
+      args: {
+        name: "file",
+        description: "The Python script to run",
+        template: "filepaths",
```
This can use the `@fig/autocomplete-generators` package to only accept `.py` files (see https://github.com/withfig/autocomplete/blob/2a99474a1ba53fd6edacb7106f9d5bcd127d2bd8/src/python3.ts#L23)

jrieke replied:
Oh that's fantastic, I was searching for that! Fixed it. Might be worth highlighting this a bit more in the docs, since I searched for this quite a while but couldn't find anything.

fedeci reviewed (changes requested):
Thanks @jrieke it looks good! I would just improve the UX more by changing that generator.

jrieke commented:
@fedeci Thanks for the quick review. I fixed the generator as you suggested. I also added emoji icons to all commands. Not sure if that's a "common" thing in Fig. We use a ton of emojis in Streamlit, so I'm sure our users will love it!

Let me know if you have any more comments!

fedeci reviewed (approved):
Thanks! Emojis are perfectly fine and are a quick and easy way to add custom icons to suggestions!
